### ISSUE TITLE
mdoc 2.12.9 & scoverage: Cannot invoke plugin without specifying <dataDir> with 

### ISSUE THREAD

**Issue by neko-kai** [sbt/sbt#4944]
Creating this issue on [request of Seth Tisue](https://github.com/7mind/izumi/commit/c9d31508ba364c0c0830b6c652e31bfe699c860f#diff-fdc3abdfd754eeb24090dbd90aeec2ceR653). This could be a bug in any of mdoc, mdoc sbt plugin or sbt, so creating the issue in both sbt and mdoc trackers. [mdoc tracker](https://github.com/scalameta/mdoc/issues/181)

To reproduce execute:

```
git clone https://github.com/7mind/izumi.git
cd izumi
git checkout bug/mdoc-scoverage-2.12.9-failure
sbt mdoc
```

Task fails with error:

```
mdoc.internal.cli.FileException: /private/tmp/izumi/doc/microsite/src/main/tut/logstage/00_logstage.md
Caused by: java.lang.RuntimeException: Cannot invoke plugin without specifying <dataDir>
	at scoverage.ScoveragePlugin.processOptions(plugin.scala:43)
	at scala.tools.nsc.plugins.Plugin.init(Plugin.scala:69)
	at scala.tools.nsc.plugins.Plugins.$anonfun$loadPlugins$9(Plugins.scala:145)
  | => mat scala.tools.nsc.plugins.Plugins.$anonfun$loadPlugins$9$adapted(Plugins.scala:145)
	at scala.collection.TraversableLike.noneIn$1(TraversableLike.scala:271)
	at scala.collection.TraversableLike.filterImpl(TraversableLike.scala:337)
	at scala.collection.TraversableLike.filterImpl$(TraversableLike.scala:249)
	at scala.collection.AbstractTraversable.filterImpl(Traversable.scala:108)
	at scala.collection.TraversableLike.filter(TraversableLike.scala:347)
	at scala.collection.TraversableLike.filter$(TraversableLike.scala:347)
	at scala.collection.AbstractTraversable.filter(Traversable.scala:108)
	at scala.tools.nsc.plugins.Plugins.loadPlugins(Plugins.scala:145)
	at scala.tools.nsc.plugins.Plugins.loadPlugins$(Plugins.scala:102)
	at scala.tools.nsc.Global.loadPlugins(Global.scala:44)
	at scala.tools.nsc.plugins.Plugins.plugins(Plugins.scala:148)
	at scala.tools.nsc.plugins.Plugins.plugins$(Plugins.scala:148)
	at scala.tools.nsc.Global.plugins$lzycompute(Global.scala:44)
	at scala.tools.nsc.Global.plugins(Global.scala:44)
	at scala.tools.nsc.plugins.Plugins.computePluginPhases(Plugins.scala:159)
	at scala.tools.nsc.plugins.Plugins.computePluginPhases$(Plugins.scala:158)
	at scala.tools.nsc.Global.computePluginPhases(Global.scala:44)
	at scala.tools.nsc.Global.computePhaseDescriptors(Global.scala:714)
	at scala.tools.nsc.Global.phaseDescriptors$lzycompute(Global.scala:719)
	at scala.tools.nsc.Global.phaseDescriptors(Global.scala:719)
	at scala.tools.nsc.Global$Run.<init>(Global.scala:1221)
	at mdoc.internal.markdown.MarkdownCompiler.compileSources(MarkdownCompiler.scala:168)
	at mdoc.internal.markdown.MarkdownCompiler.compile(MarkdownCompiler.scala:181)
	at mdoc.internal.markdown.MarkdownCompiler$.buildDocument(MarkdownCompiler.scala:44)
	at mdoc.internal.markdown.MdocPostProcessor.processScalaInputs(MdocPostProcessor.scala:136)
	at mdoc.internal.markdown.MdocPostProcessor.processDocument(MdocPostProcessor.scala:46)
	at com.vladsch.flexmark.parser.internal.PostProcessorManager.postProcess(PostProcessorManager.java:70)
	at com.vladsch.flexmark.parser.internal.PostProcessorManager.processDocument(PostProcessorManager.java:53)
	at com.vladsch.flexmark.parser.Parser.postProcess(Parser.java:389)
	at com.vladsch.flexmark.parser.Parser.parse(Parser.java:369)
	at mdoc.internal.markdown.Markdown$.toDocument(Markdown.scala:81)
	at mdoc.internal.markdown.Markdown$.toMarkdown(Markdown.scala:113)
	at mdoc.internal.cli.MainOps.handleMarkdown(MainOps.scala:64)
	at mdoc.internal.cli.MainOps.handleFile(MainOps.scala:91)
	at mdoc.internal.cli.MainOps.$anonfun$generateCompleteSite$1(MainOps.scala:130)
	at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
	at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
	at scala.collection.immutable.List.foldLeft(List.scala:89)
	at mdoc.internal.cli.MainOps.generateCompleteSite(MainOps.scala:128)
	at mdoc.internal.cli.MainOps.run(MainOps.scala:149)
	at mdoc.internal.cli.MainOps$.process(MainOps.scala:234)
	at mdoc.Main$.process(Main.scala:26)
	at mdoc.Main$.process(Main.scala:21)
	at mdoc.Main$.main(Main.scala:16)
	at mdoc.Main.main(Main.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at sbt.Run.invokeMain(Run.scala:116)
	at sbt.Run.run0(Run.scala:98)
	at sbt.Run.execute$1(Run.scala:72)
	at sbt.Run.$anonfun$run$4(Run.scala:85)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at sbt.util.InterfaceUtil$$anon$1.get(InterfaceUtil.scala:10)
	at sbt.TrapExit$App.run(TrapExit.scala:257)
	at java.base/java.lang.Thread.run(Thread.java:835)
```

Setting `scalaVersion := "2.12.8"` instead of `2.12.9` fixes this.

Note that `ScoveragePlugin` is disabled AND `coverageEna
Note that `ScoveragePlugin` is disabled AND `coverageEnabled` is set to false, yet mdoc somehow launches with scoverage compiler plugin and fails

**Comment by SethTisue** [comment]
@eed3si9n do you think there's any chance this might be a 2.12.9 regression we'd want to fix for 2.12.10?

**Comment by eed3si9n** [comment]
Just came across this via CodeTriage pick. It sounds like something might had miscalculated the scalacOptions?
It's been so long that I am not sure if this issue is still reproducible or relevant so I am going to close this, but let us know if the issue is still there.
